impl/src/main/java/org/ehcache/impl/internal/store/offheap/portability/OffHeapValueHolderPortability.java

```diff
@@ -79,7 +79,11 @@ public boolean equals(Object o, ByteBuffer byteBuffer) {
     long lastAccessTime = byteBuffer.getLong();
     long expireTime = byteBuffer.getLong();
     byteBuffer.getLong(); // hits read from disk. It is kept for compatibility reasons with previously saved data
-    return new LazyOffHeapValueHolder<>(id, byteBuffer.slice(), serializer,
+
+    ByteBuffer attached = byteBuffer.slice();
+    ByteBuffer detached = ByteBuffer.allocateDirect(attached.remaining());
```

Comment:
Are you sure about the direct byte buffer? A heap one would put less strain on the GC as it has no finalizer, and I don't see the benefit of a offheap one here.

Reply:
A heap one is probably a better choice you're right.